setup.cfg

```diff
@@ -0,0 +1,36 @@
+[metadata]
+name = qutip-cupy
+url = https://qutip.org
+description = CuPy backend for QuTiP
+long_description = file: README.md
+long_description_content_type = text/markdown; variant=GFM
+keywords = quantum, physics, dynamics, gpu, linear-algebra
+license = BSD 3-Clause License
+license_file = LICENSE.txt
+classifiers =
+    Development Status :: 2 - Pre-Alpha
+    Intended Audience :: Science/Research
+    License :: OSI Approved :: BSD License
+    Programming Language :: Python
+    Programming Language :: Python :: 3
+    Topic :: Scientific/Engineering
+    Operating System :: MacOS
+    Operating System :: POSIX
+    Operating System :: Unix
+    Operating System :: Microsoft :: Windows
+author = Alexander Pitchford, Nathan Shammah, Shahnawaz Ahmed, Neill Lambert, Eric Giguere, Boxi Li, Jake Lishman, Simon Cross
+author_email = qutip-admin@googlegroups.com
+platforms = Linux, Mac OSX, Unix, Windows
+
+[options]
+package_dir =
+    =src
```

Comment:
In this case it's because the `qutip-cupy` package is inside a larger `src` directory.  This line says "look for packages with the prefix `./src/`".  The extra `src` directory generally isn't necessary for Python packages (`qutip/qutip` doesn't have it, for example), but since we're moving to a more monolithic repository structure, it can be a small help for us keeping things neater.  It's also neater if we want to distribute more than one package from this repository (though that's not such a concern right now).

I haven't set this package up as a namespace package at the moment.  We could talk about the design in the future, but I had sort of expected that we might not want it to be automatically imported into `qutip` if it's installed. If the user has to do `import qutip_cupy; qutip_cupy.activate()` at the top of a script, there's a lot more scope for programmatic control of settings before the installation hooks itself in; `activate` can take arguments based on the system it's running on (may be useful for clusters), or you can do some setup bits and bobs before calling `activate`.  For example, if one data-layer plugin wants to interact with another, this way seems cleaner for arranging their interdependencies, and for the packages or user to arrange which implementations of the interaction functions should be used.

Reply:
Excellent. I agree with you thoughts,  and I think  it is in line with  what  @hodgestar has commented on my initial thoughts document about  having the user do some initial settings.